### ISSUE TITLE
[QUESTION] How to query JSON column

### ISSUE THREAD

**Issue by shlomi-viz** [awslabs/aws-athena-query-federation#198]
What is the best way to query JSON column? now I'm getting the error 
```
SYNTAX_ERROR: line 1:8: Column 'media_metadata' cannot be resolved
```
Can I make minior changes to my setup or query to get this info? or do I have to create a new view that will convert this JSON to text and then query the view?

Thanks

**Comment by atennak1** [comment]
Hi @shlomi-viz , could you verify which connector/database you're referring to?

**Comment by shlomi-viz** [comment]
@atennak1 any suggestions? thanks

**Comment by henrywu2019** [comment]
JSON is very useful type in Postgresql. Would like to know the answer too.

**Comment by avirtuos** [comment]
Is the type of the column 'JSON'? if so, we do not presently support that data type. We can keep this issue open as a feature request but it is also worth noting that if this is a high value feature for you we can provide guidance for you to make the code change. Depending on how you envision using the field, it may not require any changes within Athena if you can map it to an existing type.

**Comment by henrywu2019** [comment]
Yes. That is one of the reasons why we use Postgresql. Athena also supports JSON in a built-in way, so why not just map to Athena's JSON? It looks perfect at first glance. May I know anything prevents us from doing that?

**Comment by avirtuos** [comment]
It's possible but will require work on both the SDK and on the Athena side.  The SDK is open source so you can make changes there if you need this before we get to do it but changes within Athena will have to wait for our team.

**Comment by henrywu2019** [comment]
Why it needs changes on the Athena side since Athena supports JSON? I am not very familiar with the SDK. What changes are required on the SDK side?

**Comment by avirtuos** [comment]
Our SDK currently only supports a subset of Apache Arrow data types. When we add support for new/more of these types it often required changes in both the SDK and Athena. JSON would require changes on both sides.

**Comment by dnix** [comment]
Is there any update on JSON/JSONB support for postgres? Would be greatly useful for us. Thanks!

**Comment by pratikpparikh** [comment]
Is there any update on the JSON/JSONB support for postgres?
